agents/geminidataanalytics/colabs/geminidataanalytics_public_SDK_demo_|_vPublished.ipynb

```diff
@@ -0,0 +1,1538 @@
+      "source": [
+        "# @title Multi-turn Conversation\n",
+        "\n",
+        "# Re-used across requests to track previous turns\n",
+        "conversation_messages = []\n",
+        "\n",
+        "data_agent_id = \"data_agent_1\" # @param {type:\"string\"}\n",
+        "\n",
+        "data_agent_context = geminidataanalytics.DataAgentContext()\n",
+        "data_agent_context.data_agent = f\"projects/{billing_project}/locations/global/dataAgents/{data_agent_id}\"\n",
+        "# data_agent_context.credentials = credentials # uncomment this if you are using looker datasource\n",
+        "\n",
+        "# Helper function for calling the API\n",
+        "def multi_turn_Conversation(msg):\n",
+        "\n",
+        "  message = geminidataanalytics.Message()\n",
+        "  message.user_message.text = msg\n",
+        "\n",
+        "  # Send multi-turn request by including previous turns, plus new message\n",
+        "  conversation_messages.append(message)\n",
+        "\n",
+        "  request = geminidataanalytics.ChatRequest(\n",
+        "      parent=f\"projects/{billing_project}/locations/global\",\n",
+        "      messages=conversation_messages,\n",
+        "      data_agent_context=data_agent_context,\n",
+        "\n",
+        "      # uncomment the line below when using chat with inline_context\n",
+        "      # inline_context=inline_context\n",
+        "  )\n",
+        "\n",
+        "  # Make the request\n",
+        "  stream = data_chat_client.chat(request=request)\n",
+        "\n",
+        "  # Handle the response\n",
+        "  for response in stream:\n",
+        "    show_message(response)\n",
+        "    conversation_messages.append(response)"
+      ],
+      "metadata": {
+        "id": "SyBA1KHtyOgD",
+        "cellView": "form"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Send first-turn request\n",
+        "multi_turn_Conversation(\"List of the top 5 states by the total number of airports\")"
+      ],
+      "metadata": {
+        "collapsed": true,
+        "id": "pmdkDdNFzZFg"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Send follow-up-turn request\n",
+        "multi_turn_Conversation(\"Can you please make it a pie chart?\")"
+      ],
+      "metadata": {
+        "id": "WnCikHbCOy3o",
+        "collapsed": true
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Other DataAgent Service Methods"
+      ],
+      "metadata": {
+        "id": "mmz03WdUJ4Ya"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# @title Get Data Agent\n",
+        "\n",
+        "# Initialize request argument(s)\n",
+        "\n",
+        "data_agent_id = \"data_agent_1\" # @param {type:\"string\"}\n",
```

Comment:
Should this `data_agent_id` be defined as a constant so that it doesn't need to be changed everywhere?